src/main/java/gregtech/client/renderer/scene/WorldSceneRenderer.java

```diff
@@ -88,6 +94,15 @@ public WorldSceneRenderer setAfterWorldRender(Consumer<WorldSceneRenderer> callb
     public WorldSceneRenderer addRenderedBlocks(@Nullable Collection<BlockPos> blocks) {
         if (blocks != null) {
             this.renderedBlocks.addAll(blocks);
+            TILE_ENTITIES.clear();
+            blocks.forEach(pos -> {
+                TileEntity tile = world.getTileEntity(pos);
+                if (tile != null && (!(tile instanceof IGregTechTileEntity gtte) ||
+                        // Put MTEs only when it has FastRenderer
+                        gtte.getMetaTileEntity() instanceof IFastRenderMetaTileEntity)) {
+                    TILE_ENTITIES.put(pos, tile);
```

Comment:
Is there a spot where this should be cleared? Besides when recalculating the TEs?